quizler/lib.py

```diff
@@ -14,14 +15,40 @@ def get_api_envs():
     return client_id, user_id
 
 
-def api_call(end_point, client_id, user_id):
+def api_call(method: str, end_point: str, params: Dict[str, str] = None, client_id: str = None,
+             access_token: str = None):
     """Call given API end_point with API keys."""
-    url = 'https://api.quizlet.com/2.0/users/{}/{}'.format(user_id, end_point)
-    params = {'client_id': client_id}
-    response = requests.get(url, params)
-    if response.status_code != 200:
+    if bool(client_id) == bool(access_token):
+        raise ValueError('Either client_id or access_token')
+
+    url = 'https://api.quizlet.com/2.0/{}'.format(end_point)
+
+    if not params:
+        params = {}
+    if client_id:
+        params['client_id'] = client_id
+
+    headers = {'Authorization': 'Bearer {}'.format(access_token)} if access_token else None
+
+    # pylint: disable=too-many-function-args
```

Comment:
Hm, I don't get an error running `pylint` over this code even without this (and `no-member` below) comment. Do you?

Reply:
Strange, I needed them at some point. You're right, good catch! removed :)